src/vs/workbench/contrib/chat/browser/tools/projectTreeTool.ts

```diff
@@ -23,6 +23,7 @@ This tool ignores node_modules, __pycache__, dist directories, certain files lik
 This tool does not provide information for specific files or directories, but rather gives an overview of the entire project structure.
 This tool is helpful for locating files and directories in the workspace.
 This tool only needs to be called once per conversation, unless files or directories are added, removed, moved, or renamed in the workspace.
+If the workspace is empty or no workspace folders are open, this tool cannot generate a project tree.
```

Comment:
Another approach is to include workspace information in the context message and instruct the model not to call the tool if the context shows that there is no workspace open. I was able to get Anthropic to do less inspect variables requests for non-existent variables in that way:

> ...
> 
> If the user references a variable by name, first determine the `access_key` from the user context or a previous inspect variables result.
> 
> Do not call when:
>   - the variables do not appear in the user context
> ...

Reply:
Also, my intuition is that more direct instructions work better (and use fewer tokens) i.e.

> Do not use this tool when the workspace is empty or no workspace is open.

Reply:
Ah this is much nicer! I rebased to pull in the context changes and was able to add a note to the context when workspace folders are not open. Now, with a corresponding update to the workspace tools descriptions, anthropic doesn't call any of the workspace tools if we've noted in the context that workspace folders aren't open 👌